app/javascript/packs/carts.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+var csrf_token = $('meta[name="csrf-token"]').attr('content');
+$(".add-cart").click(function(){
+    console.log("aaaa");
+    let product_id = $(this).attr("data-product-id");
+    $.ajax({
+      url: '/carts',
+      type: 'POST',
+      headers: {
+        'X-CSRF-Token': csrf_token,
```

Comment:
nếu k có cái này thì sao e

Reply:
theo em tìm hiểu cái này nó nhận diện user quan token. để tránh tấn công sử dụng quyền của người dùng đấy. ví dụ như hacker biết các phương thức mà user đấy có quyền thực hiện: thêm xóa rỏ hàng...
k có vẫn được nhưng hoàn toàn có thể bị tấn công cfrf nên tránh các lỗi tấn công phổ biến này ạ